File: python/lv1/소수찾기.py
N = 1000000
prime_nums = [1] * (N+2)
prime_nums[0] = 0
prime_nums[1] = 0

for i in range(2, (N+1)//2):
    for j in range(2*i, N+1, i):
        if prime_nums[j]:
            prime_nums[j] = 0

def solution(n):
    return sum(prime_nums[:n+1])
    
    

n = [10, 5]
result = [4, 3]

for i in range(len(n)):
    if solution(n[i]) == result[i]:
        print(f'{i}: pass')
    else:
        print(f'{i}: fail')

File: js/lv1/소수찾기.js
const N = 1000000;
let prime_nums = new Array(N + 1).fill(true);
prime_nums[0] = false;
prime_nums[1] = false;

for (let i = 2; i < parseInt((N + 1) / 2); i++) {
  for (let j = 2 * i; j < N + 1; j = j + i) {
    if (prime_nums[j] == true) {
      prime_nums[j] = false;
    }
  }
}

function solution(n) {
  const range = prime_nums.slice(0, n + 1);
  let answer = 0;
  range.forEach((check) => {
    if (check === true) {
      answer++;
    }
  });
  return answer;
}

n = [10, 5];
result = [4, 3];

for (let i = 0; i < n.length; i++) {
  if (solution(n[i]) === result[i]) {
    console.log(`${i}: pass`);
  } else {
    console.log(`${i}: fail`);
  }
}
